src/concrete/ml/onnx/ops_impl.py

```diff
@@ -897,7 +899,36 @@ def numpy_equal(
     Returns:
         Tuple[numpy.ndarray]: Output tensor
     """
+    return (numpy.equal(x, y),)
+
+
+def rounded_numpy_equal_for_trees(
+    x: numpy.ndarray,
+    y: numpy.ndarray,
+    *,
+    lsbs_to_remove_for_trees: Optional[int] = None,
+) -> Tuple[numpy.ndarray]:
+    """Compute equal in numpy according to ONNX spec.
+
+    See https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Equal-11
+
+    Args:
+        x (numpy.ndarray): Input tensor
+        y (numpy.ndarray): Input tensor
+        lsbs_to_remove_for_trees (Optional[int]): Number of the least significant bits to remove
+            for tree-based models only.
+
+    Returns:
+        Tuple[numpy.ndarray]: Output tensor
+    """
+
+    # For tree-based models, x == y <=> x <= y or x < y - 1, because y is the max sum.
+    if lsbs_to_remove_for_trees is not None and lsbs_to_remove_for_trees > 0:
+        return rounded_comparison(
+            y, x, lsbs_to_remove_for_trees, operation=lambda x: x >= 0
```

Comment:
I would at least explain here (+ in LessOrEqual) or in `rounded_comparison` why this order is important (ie, why the function is not symmetric on x,y mostly) to be sure we understand it correctly in the future !

Reply:
this part will be deleted once we add auto truncate